generator/src/main/java/com/scottlogic/deg/generator/generation/fieldvaluesources/DateTime/TemporalFieldValueSource.java

```diff
@@ -65,17 +49,16 @@ public boolean isFinite() {
     public long getValueCount() {
 
         if (isFinite()) {
-            ChronoUnit unit = getIncrementForDuration(inclusiveLower, exclusiveUpper);
-
             Duration duration = Duration.between(inclusiveLower, exclusiveUpper);
             Period period = Period.between(inclusiveLower.toLocalDate(), exclusiveUpper.toLocalDate());
 
-            if (unit == ChronoUnit.SECONDS) return duration.getSeconds();
-            if (unit == ChronoUnit.MINUTES) return duration.getSeconds() * 60;
-            if (unit == ChronoUnit.HOURS) return duration.getSeconds() * 60 * 60;
-            if (unit == ChronoUnit.DAYS) return (period.getDays() + 1);
-            if (unit == ChronoUnit.MONTHS) return (period.getMonths() + 1);
-            if (unit == ChronoUnit.YEARS) return (period.getYears() + 1);
+            if (granularity == ChronoUnit.MILLIS) return (duration.getNano() / 1000000) +1;
+            if (granularity == ChronoUnit.SECONDS) return duration.getSeconds() + 1;
```

Comment:
This is all unreachable code right now? Comment to mention further work is expected here